torch/onnx/__init__.py

```diff
@@ -454,10 +115,24 @@ def symbolic_override_first_arg_based(symbolic_fn):
     needs to be a Variable.
     """
 
-    return functools.partial(_symbolic_override_wrapper_maker, symbolic_fn, True)
+    def might_trace(args):
+        import torch
+        return torch._C._jit_is_tracing(args[0])
+
+    return functools.partial(_symbolic_override_wrapper_maker, symbolic_fn, might_trace)
+
+
+def symbolic_override_packed_sequence_based(symbolic_fn):
```

Comment:
Can we get rid of all those `*_based` decorators and just always use `torch.jit._tracing` to decide which path to take? It's probably even cheaper than using the first arg.

Reply:
probably, but I think that's a separate cleanup